arm-apimanagement/2016-07-07/swagger/apimdeployment.json

```diff
@@ -0,0 +1,1236 @@
+        "parameters": [
+          {
+            "name": "sourceResourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "Source resource group name.",
+            "pattern": "^[-\\w\\._\\(\\)]+$",
+            "minLength": 1,
+            "maxLength": 90
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ResourcesMoveInfo"
+            },
+            "description": "move resources' parameters."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {         
+          "204": {
+            "description": "Successfully moved ApiManagement resources."
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/managedeployments": {
+      "post": {
+        "tags": [
+          "ResourceProvider"
+        ],
+        "operationId": "ApiManagementService_ManageDeployments",
+        "description": "Manage deployments of an Api Management service. This operation can be used to  Change Sku or Sku Units, Change Service Tier (Developer/Standard/Premium) and Manage Vpn Configuration. This is a long running operation and could take several minutes to complete.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ServiceNameParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ApiServiceManageDeploymentsParameters"
+            },
+            "description": "Parameters supplied to ManageDeployments operation."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully applied the new deployment Configuration on the ApiManagement service.",
+            "schema": {
+              "$ref": "#/definitions/ApiServiceResource"
+            }
+          },
+          "400": {
+            "description": "Request valdiation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          },
+          "404": {
+            "description": "The service does not exist.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          },
+          "409": {
```

Comment:
Explicitly defining negative error codes will make the generated code not throw on those error codes. It will deserialize the responses based on the schema and will return the object. I believe, even the return type of the method will we be an Object as there is nothing common between ApiServiceResource and ErrorResponse.
Please define a "default" response and it's schema should be Error Response. 

Please read the default and the negative response section over here: https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/creating-swagger.md#default-response